### ISSUE TITLE
Add a timeout to deletions.

### ISSUE THREAD

**Issue by davidby-influx** [influxdata/influxdb#23164]
Some customers are submitting hundreds of concurrent deletions.  Concurrent deletions are [limited by a configuration setting](https://github.com/influxdata/influxdb/blob/7d182158f436e454c27af01cc90cbd2e9c759923/tsdb/store.go#L1396), so when a large number (we have seen over 600 concurrently) are submitted, it would be good to have a timeout for them analogous to the write or query timeouts.

This would require adding a new configuration parameter, and modifying `Store.DeleteSeries` to use a timer.

**Comment by lesam** [comment]
Makes sense as a service protection measure, as long as timed out deletes actually release resources - we don't want to time out to the user and still have a long-running goroutine in the background trying to do the delete. Especially if the deletes are retried.
